From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AlphaKR93 <alphakr93@outlook.com>
Date: Sun, 3 Jul 2022 22:52:31 +0900
Subject: [PATCH] JettPack API Patches

Original by Simon Gardling <titaniumtown@gmail.com>

Copyright (C) JettPackMC LLC

This program is free software: you can redistribute it and/or modify it
under the terms of the GNU General Public License as published by the
Free Software Foundation, either version 3 of the License, or (at your
option) any later version.

This program is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
General Public License for more details.

You should have received a copy of the GNU General Public License along
with this program. If not, see <http://www.gnu.org/licenses/>.

[ADDED PATCHES]
0001 - skip event if no listeners

diff --git a/src/main/java/org/bukkit/plugin/SimplePluginManager.java b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
index 8f3227c005a4c54dd3db530b46eef8be67865afa..d4b8cd5e8fc82855a33d2f749e6ca6054f8ff388 100644
--- a/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+++ b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
@@ -665,6 +665,11 @@ public final class SimplePluginManager implements PluginManager {
      */
     @Override
     public void callEvent(@NotNull Event event) {
+        // Prismarine start - JettPack patches
+        HandlerList handlers = event.getHandlers();
+        RegisteredListener[] listeners = handlers.getRegisteredListeners();
+        if (listeners.length == 0) return;
+        // Prismarine end
         // Paper - replace callEvent by merging to below method
         if (event.isAsynchronous() && server.isPrimaryThread()) {
             throw new IllegalStateException(event.getEventName() + " may only be triggered asynchronously.");
@@ -672,8 +677,8 @@ public final class SimplePluginManager implements PluginManager {
             throw new IllegalStateException(event.getEventName() + " may only be triggered synchronously.");
         }
 
-        HandlerList handlers = event.getHandlers();
-        RegisteredListener[] listeners = handlers.getRegisteredListeners();
+        // HandlerList handlers = event.getHandlers(); // JettPack - moved up // Prismarine - JettPack patches
+        // RegisteredListener[] listeners = handlers.getRegisteredListeners(); // JettPack - moved up // Prismarine - JettPack patches
 
         for (RegisteredListener registration : listeners) {
             if (!registration.getPlugin().isEnabled()) {
